File: RagnarokBotWeb/ClientApp/src/app/models/player.dto.ts
import { ScumServer } from "./scum-server";

export class PlayerDto {
    id!: number;
    name?: string;
    scumId?: string;
    steamId64?: string;
    steamName?: string;
    discordId?: string;
    discordName?: string;
    scumServer?: ScumServer;
    money?: number;
    gold?: number;
    fame?: number;
    coin?: number;
    createDate?: number;
    isVip!: boolean;
    isBanned!: boolean;
    isSilenced!: boolean;
    vipExpiresAt?: Date;
    banExpiresAt?: Date;
    silenceExpiresAt?: Date;
    lastLoggedIn?: Date;
}

File: RagnarokBotWeb/Domain/Services/Dto/PlayerDto.cs
namespace RagnarokBotWeb.Domain.Services.Dto
{
    public class PlayerDto
    {
        public long Id { get; set; }
        public string? Name { get; set; }
        public string? ScumId { get; set; }
        public string? SteamId64 { get; set; }
        public string? SteamName { get; set; }
        public string? DiscordId { get; set; }
        public string? DiscordName { get; set; }
        public ScumServerDto ScumServer { get; set; }
        public long? Money { get; set; }
        public long? Gold { get; set; }
        public long? Fame { get; set; }
        public long Coin { get; set; } = 0;
        public DateTime CreateDate { get; set; }
    }
}
